client/src/components/Upload/Upload.js

```diff
@@ -0,0 +1,247 @@
+            position: absolute;
+            top: calc(50% - 50px);
+            left: 50%;
+        }
+
+        .loading-dot {
+            height: 5px;
+            width: 5px;
+            background-color: white;
+            border-radius: 100%;
+            opacity: 0;
+        }
+
+        .loading-dot-animated {
+            animation-name: loading-dot-animated;
+            animation-direction: alternate;
+            animation-duration: .75s;
+            animation-iteration-count: infinite;
+            animation-timing-function: ease-in-out;
+        }
+
+        @keyframes loading-dot-animated {
+            from {
+                opacity: 0;
+            }
+
+            to {
+                opacity: 1;
+            }
+        }
+    `
+  document.body.appendChild(style);
+  var frame = document.createElement("div");
+  frame.id = "divLoadingFrame";
+  frame.classList.add("loading-frame");
+  for (var i = 0; i < 10; i++) {
+    var track = document.createElement("div");
+    track.classList.add("loading-track");
+    var dot = document.createElement("div");
+    dot.classList.add("loading-dot");
+    track.style.transform = "rotate(" + String(i * 36) + "deg)";
+    track.appendChild(dot);
+    frame.appendChild(track);
+  }
+  document.body.appendChild(frame);
+  var wait = 0;
+  var dots = document.getElementsByClassName("loading-dot");
+  // There was a warning i is already defined, thus use j for index instead.
+  for (var j = 0; j < dots.length; j++) {
+    window.setTimeout(function(dot) {
+      dot.classList.add("loading-dot-animated");
+    }, wait, dots[j]);
+    wait += 150;
+  }
+};
+
+function removeLoading() {
+  document.body.removeChild(document.getElementById("divLoadingFrame"));
+  document.body.removeChild(document.getElementById("styleLoadingWindow"));
+};
+
+document.addEventListener('keydown', function(e) {
+  if (e.keyCode === 27) {
+    removeLoading();
+  }
+}, false);
+
+function Upload() {
+  const [user, loading] = useAuthState(auth);
+  const [uploadedFiles, setUploadedFiles] = useState([]); // This is the array of files that have been uploaded.
+  const [title] = React.useState();
+  const history = useHistory();
+  const [uploadedSuccessfully, setUploadedSuccessfully] = useState(false);
+  useEffect(() => {
+    if (loading) {
+      // maybe trigger a loading screen
+      return;
+    }
+    if (!user) history.replace("/home");
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [user, loading]);
+  
+  // Select audio file to upload.
+  const submitAudios = async (files) => {
+    //for each file in the array of files, asynchronously upload the file to the server.
+    const token = await auth.currentUser.getIdToken();
+    showLoading();
+    let formData = new FormData();
+    // wait till all files are appended to the form data before sending it to the server.
+    for (let i = 0; i < files.length; i++) {
+      formData.append("file", files[i]);
+    }
```

Comment:
```suggestion
    files.forEach((f) => formData.append("file", f));
```
just a suggestion (not sure if the syntax is fully correct since using github, check it before committing the change)